File: idle-kluch-server/src/main/java/com/soze/idlekluch/game/engine/nodes/Nodes.java
package com.soze.idlekluch.game.engine.nodes;

import com.soze.idlekluch.game.engine.components.*;
import com.soze.idlekluch.game.engine.components.resourceharvester.ResourceHarvesterComponent;
import com.soze.klecs.node.Node;

public interface Nodes {

  Node PHYSICAL_NODE = Node.of(PhysicsComponent.class);
  Node OWNERSHIP = Node.of(OwnershipComponent.class);
  Node BUILDABLE = Node.of(BuildableComponent.class);
  Node RESOURCE_SOURCE = Node.of(ResourceSourceComponent.class);
  Node OCCUPY_SPACE = Node.of(PhysicsComponent.class, StaticOccupySpaceComponent.class);
  Node STORAGE = Node.of(PhysicsComponent.class, ResourceStorageComponent.class);

  Node BUILDING = Node.of(PhysicsComponent.class, OwnershipComponent.class, BuildableComponent.class, StaticOccupySpaceComponent.class);

  Node HARVESTER = Node.of(ResourceHarvesterComponent.class, ResourceStorageComponent.class);
  Node SELLER = Node.of(ResourceStorageComponent.class, ResourceSellerComponent.class);

}

File: idle-kluch-client/src/ecs/nodes.js
import Node from "./Node";
import { PhysicsComponent } from "./components/PhysicsComponent";
import { NameComponent } from "./components/NameComponent";
import { GraphicsComponent } from "./components/GraphicsComponent";
import { ResourceHarvesterComponent } from "./components/ResourceHarvesterComponent";
import { ResourceStorageComponent } from "./components/ResourceStorageComponent";

export const NAME_RENDER_NODE = Node.of(PhysicsComponent, NameComponent);
export const GRAPHICS_RENDER_NODE = Node.of(PhysicsComponent, GraphicsComponent);
export const RESOURCE_HARVESTER_NODE = Node.of(PhysicsComponent, ResourceHarvesterComponent, ResourceStorageComponent);
export const STORAGE_NODE = Node.of(PhysicsComponent, ResourceStorageComponent);